code/__HELPERS/unsorted.dm

```diff
@@ -1584,4 +1584,22 @@ Game Mode config tags:
 	playsound(T, fire_sound, 50, 1)
 	A.OnFired()
 	spawn()
-		A.process()
\ No newline at end of file
+		A.process()
+
+
+//Increases delay as the server gets more overloaded,
+//as sleeps aren't cheap and sleeping only to wake up and sleep again is wasteful
+#define DELTA_CALC max(((max(world.tick_usage, world.cpu) / 100) * max(Master.sleep_delta,1)), 1)
+
+/proc/stoplag()
+	. = round(1*DELTA_CALC)
+	sleep(world.tick_lag)
+	if (world.tick_usage > TICK_LIMIT_TO_RUN) //woke up, still not enough tick, sleep for more.
+		. += round(2*DELTA_CALC)
+		sleep(world.tick_lag*2*DELTA_CALC)
+		if (world.tick_usage > TICK_LIMIT_TO_RUN) //woke up, STILL not enough tick, sleep for more.
+			. += round(4*DELTA_CALC)
+			sleep(world.tick_lag*4*DELTA_CALC)
+			//you might be thinking of adding more steps to this, or making it use a loop and a counter var
```

Comment:
We don't actually have explosions that throw stuff on here so I doubt that's applicable.